### PR TITLE
Moove webhook publish events

### PR DESCRIPTION
**Pull request by barbararochazup**
**Notify Hermes module when:**
Deploy started and finished (with success or not)
Undeploy started and finished (with success or not)

### CODE REVIEW

File: moove/application/src/main/kotlin/io/charlescd/moove/application/WebhookEventService.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        )
+    }
+
+    private fun getBuild(id: String): Build {
+        return this.buildRepository.findById(
+            id
+        ).orElseThrow {
+            NotFoundException("build", id)
+        }
+    }
```
Comment (emiteze) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/WebhookEventService.kt:99]: Use find from our BuildService would remove the need of this code

File: moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/request/DeploymentCallbackRequest.kt [hunk 2]
```diff
@@ -24,8 +24,16 @@ data class DeploymentCallbackRequest(
 ) {
 
     fun isCallbackStatusSuccessful() = this.deploymentStatus == DeploymentRequestStatus.SUCCEEDED
+
+    fun isDeployEvent() = listOf(DeploymentRequestStatus.SUCCEEDED, DeploymentRequestStatus.FAILED).contains(this.deploymentStatus)
+
+    fun isUndeployEvent() = listOf(DeploymentRequestStatus.UNDEPLOYED, DeploymentRequestStatus.UNDEPLOYED).contains(this.deploymentStatus)
```
Comment (emiteze) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/request/DeploymentCallbackRequest.kt:30]: Duplicated DeploymentRequestStatus.UNDEPLOYED

File: moove/infrastructure/src/main/kotlin/io/charlescd/moove/infrastructure/service/client/HermesPublisherClient.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+package io.charlescd.moove.infrastructure.service.client
+
+import io.charlescd.moove.infrastructure.configuration.HermesPublisherEncoderConfiguration
+import io.charlescd.moove.infrastructure.service.client.request.HermesPublishSubscriptionEventRequest
+import org.springframework.cloud.openfeign.FeignClient
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.*
+
+@FeignClient(name = "hermesClient", url = "\${charlescd.hermes.url}", configuration = [ HermesPublisherEncoderConfiguration::class])
```
Comment (emiteze) [inline moove/infrastructure/src/main/kotlin/io/charlescd/moove/infrastructure/service/client/HermesPublisherClient.kt:26]: Can we remove this blank space inside of this configuration list?

**Review by emiteze — CHANGES_REQUESTED** [review]

**Review by pedroborgeszup — APPROVED** [review]

File: moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/request/DeploymentCallbackRequest.kt (hunk 2 above)
Comment (barbararochazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/request/DeploymentCallbackRequest.kt:30]: This method was not called, so I removed them

File: moove/infrastructure/src/main/kotlin/io/charlescd/moove/infrastructure/service/client/HermesPublisherClient.kt (hunk 3 above)
Comment (barbararochazup) [reply moove/infrastructure/src/main/kotlin/io/charlescd/moove/infrastructure/service/client/HermesPublisherClient.kt:26]: Done

File: moove/application/src/main/kotlin/io/charlescd/moove/application/WebhookEventService.kt (hunk 1 above)
Comment (barbararochazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/WebhookEventService.kt:99]: Done

**Review by emiteze — APPROVED** [review]

**Pull request merged**